### ISSUE TITLE
The kernel appears to have died. It will restart automatically, when task_type='GPU' and loss_function = 'MultiLogloss

### ISSUE THREAD

**Issue by xaeliudzyh** [catboost/catboost#3038]
Problem: Got this error message on Jupyter Notebook when using GPU.
CatBoost version: 1.2.7
GPU: NVIDIA A100 80GB PCIe
Python: 3.10.11
```
from catboost import CatBoostClassifier, Pool
from catboost.utils import eval_metric

train_pool = Pool(X_train, y_train)
val_pool = Pool(X_val, y_val)

cb = CatBoostClassifier(gpu_ram_part=0.5 ,iterations = 1000, learning_rate = 0.03,
                        max_depth = 11, loss_function = 'MultiLogloss', eval_metric = 'MultiLogloss', task_type = 'GPU', devices = '0')

cb.fit(train_pool, eval_set = val_pool, plot = True, verbose = 50, early_stopping_rounds = 50)
```
Sometimes trains works fine, sometimes I get this error. Don't know what can cause it.

I would be grateful for advice.

**Comment by andrey-khropov** [comment]
What is your OS?

How many samples do train and eval datasets contain? How many features?

> max_depth = 11

Are you sure that you need such depth? Big tree depth in the case of oblivious trees could be problematic (every depth increase multiplies the number of leaves in each tree by 2)

**Comment by xaeliudzyh** [comment]
> What is your OS?
> 
> How many samples do train and eval datasets contain? How many features?
> 
> > max_depth = 11
> 
> Are you sure that you need such depth? Big tree depth in the case of oblivious trees could be problematic (every depth increase multiplies the number of leaves in each tree by 2)


train_data - 139523 samples
val_data - 24621 samples
features - 1024(vector representation of text given by BGE-M3)

I did train with max_depth = 9, and judging by the training process, I wanted to try a deeper depth.

**Comment by andrey-khropov** [comment]
> > What is your OS?

up

> > How many samples do train and eval datasets contain? How many features?
> > > max_depth = 11
> > 
> > 
> > Are you sure that you need such depth? Big tree depth in the case of oblivious trees could be problematic (every depth increase multiplies the number of leaves in each tree by 2)
> 
> train_data - 139523 samples val_data - 24621 samples features - 1024(vector representation of text given by BGE-M3)
> 
> I did train with max_depth = 9, and judging by the training process, I wanted to try a deeper depth.

How many different labels do your targets have?

I also suggest to try the latest CatBoost version (1.2.10).

**Comment by xaeliudzyh** [comment]
>  > What is your OS?

OS - Windows


> How many different labels do your targets have?

30 labels for targets
> I also suggest to try the latest CatBoost version (1.2.10).

thx for advice, i'll try latest version
